contrib/install.sh

```diff
@@ -0,0 +1,1121 @@
+
+  while true; do
+    # Prompt user for proxy input
+    proxy=$(dialog --title "Floresta-Installer (Set Proxy)" \
+                  --inputbox "Provide a proxy address (Optional: http://, https://, or socks5:// followed by IPv4, IPv6, or domain:port)" \
+                  10 60 \
+                  3>&1 1>&2 2>&3)
+
+    # Check if user pressed Cancel or ESC
+    if [ $? -ne 0 ]; then
+      dialog --title "Floresta-Installer" --msgbox "❌ Proxy input canceled. Returning to setup menu." 8 45
+      interactive_setup
+      return 1
+    fi
+
+    # Validate proxy format
+    if [[ "$proxy" =~ $proxy_regex ]]; then
+      dialog --title "Floresta-Installer" --msgbox "✅ Proxy set successfully: $proxy" 10 60
+      interactive_setup
+      return 0
+    else
+      dialog --title "Invalid Proxy" --msgbox "❌ The proxy address you entered is invalid.\n\nIt should be in the format:\n- http://IPv4:port\n- https://IPv6:port\n- socks5://IPv4:port\n- IPv4:port\n- [IPv6]:port\n\nExample: 192.168.1.100:8080 or [2001:db8::1]:9050\n\nReturning to setup menu." 12 60
+    fi
+  done
+}
+
+# func: interactive_connect
+#
+# This function setup a node to connect when daemon starts
+interactive_connect() {
+  local connect_regex="^((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9]?[0-9])(:([0-9]{1,5}))?$"
+
+  while true; do
+    # Prompt user for proxy input
+    connect=$(dialog --title "Floresta-Installer (Set a node to connect)" \
+                  --inputbox "We'll connect ONLY to this node. It should be an ipv4 address in the format <address>[:<port>]." \
+                  10 60 \
+                  3>&1 1>&2 2>&3)
+
+    # Check if user pressed Cancel or ESC
+    if [ $? -ne 0 ]; then
+      dialog --title "Floresta-Installer" --msgbox "❌ Connect input canceled. Returning to setup menu." 8 45
+      interactive_setup
+      return 1
+    fi
+
+    # Validate proxy format
+    if [[ "$proxy" =~ $proxy_regex ]]; then
+      dialog --title "Floresta-Installer" --msgbox "✅ A node to connect to set successfully: $connect" 10 60
+      interactive_setup
+      return 0
+    else
+      dialog --title "Invalid node to connect" --msgbox "❌ The node address you entered is invalid.\n\nIt should be in the format:\n- http://IPv4:port\n- https://IPv6:port\n- socks5://IPv4:port\n- IPv4:port\n- [IPv6]:port\n\nExample: 192.168.1.100:8080 or [2001:db8::1]:9050\n\nReturning to setup menu." 12 60
+    fi
+  done
+}
+
+# func: interactive_zeromq
+#
+# This function setup an address for the ZeroMQ server to listen to if user wants add one
+interactive_zeromq() {
+  local zeromq_regex="^((http|https):\/\/)?((([0-9]{1,3}\.){3}[0-9]{1,3})|\[?[a-fA-F0-9:]+\]?):[0-9]{1,5}$"
+
+  while true; do
+    # Prompt user for zeromq input
+    zeromq=$(dialog --title "Floresta-Installer (Set zeromq)" \
+                  --inputbox "Provide a zeromq address (Optional: http://domain:port, https://domain:port, IPv4:port or IPv6:port)" \
```

Comment:
ZMQ is a lightweight and efficient message queue system used for Inter Process Communication. Floresta allows you to use this to get notified about new blocks. This option sets the socket that floresta will listen for ZMQ subscribers.